src/lammpsio/snapshot.py

```diff
@@ -176,31 +222,122 @@ def to_hoomd_gsd(self, type_map=None):
                 for typeidx, typeid in enumerate(sorted_typeids):
                     frame.particles.typeid[self.typeid == typeid] = typeidx
             else:
-                frame.particles.types = list(type_map.values())
-                reverse_type_map = {
-                    typeid: typeidx for typeidx, typeid in enumerate(type_map.keys())
-                }
-                for i, typeid in enumerate(self.typeid):
-                    frame.particles.typeid[i] = reverse_type_map[typeid]
+                if type_map is not None:
+                    warnings.warn(
+                        "type_map will be deprecated, use label instead.",
+                        DeprecationWarning,
+                        stacklevel=2,
+                    )
+                    type_label_map = LabelMap(map=type_map)
+                else:
+                    type_label_map = self.label
+                frame.particles.types = type_label_map.types
+                Snapshot._set_type_id(
+                    self,
```

Comment:
Is there a reason this doesn't work?
```suggestion
                self._set_type_id(
```

If you aren't really using any class variables inside this method, you could also define it outside the class as a private helper function.

Reply:
I think you're write that a private helper function would be best here to not clutter the class.